From 3cc155296f63956ddb11ded006289a18311b1ae8 Mon Sep 17 00:00:00 2001
From: Punit Soni <punits@codeaurora.org>
Date: Wed, 6 May 2015 18:13:40 -0700
Subject: [PATCH 4/9] camera-hal: reduce logging in camera app

- remove stderr logs
- use syslog for app logging
- disable verbose logs
---
 QCamera/stack/mm-camera-interface/inc/mm_camera_dbg.h  |  4 ++--
 QCamera2/stack/mm-camera-interface/inc/mm_camera_dbg.h | 15 +++++----------
 QCamera2/stack/mm-camera-test/inc/mm_qcamera_dbg.h     | 15 ++++++++-------
 QCamera2/stack/mm-camera-test/src/mm_qcamera_preview.c |  4 ++--
 QCamera2/stack/mm-jpeg-interface/inc/mm_jpeg_dbg.h     | 13 +++++++------
 mm-image-codec/qomx_core/qomx_core.c                   |  3 ++-
 qcamlib/test/qcamlib_test.c                            | 15 +++++++++++++++
 7 files changed, 41 insertions(+), 28 deletions(-)

diff --git a/QCamera/stack/mm-camera-interface/inc/mm_camera_dbg.h b/QCamera/stack/mm-camera-interface/inc/mm_camera_dbg.h
index fcb7028..3bdfe6f 100755
--- a/QCamera/stack/mm-camera-interface/inc/mm_camera_dbg.h
+++ b/QCamera/stack/mm-camera-interface/inc/mm_camera_dbg.h
@@ -56,7 +56,7 @@ IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
     #define CDBG(fmt, args...) ALOGE(fmt, ##args)
   #else
     #include <stdio.h>
-    #define CDBG(fmt, args...) fprintf(stderr, fmt, ##args)
+    #define CDBG(fmt, args...) do {} while (0)
     #define ALOGE(fmt, args...) fprintf(stderr, fmt, ##args)
   #endif
 #endif
@@ -65,7 +65,7 @@ IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
   #define CDBG_HIGH(fmt, args...)  ALOGE(fmt, ##args)
   #define CDBG_ERROR(fmt, args...)  ALOGE(fmt, ##args)
 #else
-  #define CDBG_HIGH(fmt, args...) fprintf(stderr, fmt, ##args)
+  #define CDBG_HIGH(fmt, args...) do {} while (0)
   #define CDBG_ERROR(fmt, args...) fprintf(stderr, fmt, ##args)
 #endif
 #endif /* __MM_CAMERA_DBG_H__ */
diff --git a/QCamera2/stack/mm-camera-interface/inc/mm_camera_dbg.h b/QCamera2/stack/mm-camera-interface/inc/mm_camera_dbg.h
index fffa7e7..3099178 100755
--- a/QCamera2/stack/mm-camera-interface/inc/mm_camera_dbg.h
+++ b/QCamera2/stack/mm-camera-interface/inc/mm_camera_dbg.h
@@ -55,7 +55,7 @@
     #include <utils/Log.h>
     #define CDBG(fmt, args...) ALOGE(fmt, ##args)
   #else
-    #include <stdio.h>
+    #include <syslog.h>
     #define CDBG(fmt, args...) fprintf(stderr, fmt"\n", ##args)
     #define ALOGE(fmt, args...) fprintf(stderr, fmt"\n", ##args)
   #endif
@@ -65,15 +65,10 @@
   #define CDBG_HIGH(fmt, args...)  ALOGD(fmt, ##args)
   #define CDBG_ERROR(fmt, args...)  ALOGE(fmt, ##args)
 #else
-  #define CDBG_HIGH(fmt, args...) fprintf(stderr, fmt"\n", ##args)
-  #define CDBG_ERROR(fmt, args...) fprintf(stderr, fmt"\n", ##args)
+  #include <syslog.h>
+  #define CDBG(fmt, args...) do{}while(0)
+  #define CDBG_HIGH(fmt, args...) syslog(LOG_INFO, fmt, ##args)
+  #define CDBG_ERROR(fmt, args...) syslog(LOG_ERR, fmt, ##args)
 #endif
 
-#undef CDBG_HIGH
-#undef CDBG_ERROR
-
-#define CDBG(fmt, args...) fprintf(stderr, fmt"\n", ##args)
-#define CDBG_HIGH(fmt, args...) fprintf(stderr, fmt"\n", ##args)
-#define CDBG_ERROR(fmt, args...) fprintf(stderr, fmt"\n", ##args)
-
 #endif /* __MM_CAMERA_DBG_H__ */
diff --git a/QCamera2/stack/mm-camera-test/inc/mm_qcamera_dbg.h b/QCamera2/stack/mm-camera-test/inc/mm_qcamera_dbg.h
index d38c1b7..d424287 100644
--- a/QCamera2/stack/mm-camera-test/inc/mm_qcamera_dbg.h
+++ b/QCamera2/stack/mm-camera-test/inc/mm_qcamera_dbg.h
@@ -30,7 +30,7 @@
 #ifndef __MM_QCAMERA_DBG_H__
 #define __MM_QCAMERA_DBG_H__
 
-#define LOG_DEBUG 1
+#define LOG_DEBUG 0
 
 #ifndef LOG_DEBUG
   #ifdef _ANDROID_
@@ -55,9 +55,9 @@
     #include <utils/Log.h>
     #define CDBG(fmt, args...) ALOGE(fmt, ##args)
   #else
-    #include <stdio.h>
-    #define CDBG(fmt, args...) fprintf(stderr, fmt"\n", ##args)
-    #define ALOGE(fmt, args...) fprintf(stderr, fmt"\n", ##args)
+    #include <syslog.h>
+    #define CDBG(fmt, args...) do {} while (0)
+    #define ALOGE(fmt, args...) syslog(LOG_ERR, fmt, ##args)
   #endif
 #endif
 
@@ -65,9 +65,10 @@
   #define CDBG_HIGH(fmt, args...)  ALOGE(fmt, ##args)
   #define CDBG_ERROR(fmt, args...)  ALOGE(fmt, ##args)
 #else
-  #define CDBG(fmt, args...) fprintf(stderr, fmt"\n", ##args)
-  #define CDBG_HIGH(fmt, args...) fprintf(stderr, fmt"\n", ##args)
-  #define CDBG_ERROR(fmt, args...) fprintf(stderr, fmt"\n", ##args)
+  #include <syslog.h>
+  #define CDBG(fmt, args...) do {} while (0)
+  #define CDBG_HIGH(fmt, args...) do {} while (0)
+  #define CDBG_ERROR(fmt, args...) syslog(LOG_ERR, fmt, ##args)
 #endif
 
 #define DUMP_PRV_IN_FILE 0
diff --git a/QCamera2/stack/mm-camera-test/src/mm_qcamera_preview.c b/QCamera2/stack/mm-camera-test/src/mm_qcamera_preview.c
index aea95f4..c9009be 100644
--- a/QCamera2/stack/mm-camera-test/src/mm_qcamera_preview.c
+++ b/QCamera2/stack/mm-camera-test/src/mm_qcamera_preview.c
@@ -114,7 +114,7 @@ static void mm_app_preview_notify_cb(mm_camera_super_buf_t *bufs,
     mm_camera_buf_def_t *frame = bufs->bufs[0];
     mm_camera_test_obj_t *pme = (mm_camera_test_obj_t *)user_data;
 
-    CDBG_ERROR("%s: BEGIN - length=%d, frame idx = %d\n",
+    CDBG("%s: BEGIN - length=%d, frame idx = %d\n",
          __func__, frame->frame_len, frame->frame_idx);
 
     /* find channel */
@@ -161,7 +161,7 @@ static void mm_app_preview_notify_cb(mm_camera_super_buf_t *bufs,
 	frame_count++;
 #endif
     if (pme->user_preview_cb) {
-        CDBG_ERROR("[DBG] %s, user defined own preview cb. calling it...", __func__);
+        CDBG("[DBG] %s, user defined own preview cb. calling it...", __func__);
         pme->user_preview_cb(frame);
     }
     if (MM_CAMERA_OK != pme->cam->ops->qbuf(bufs->camera_handle,
diff --git a/QCamera2/stack/mm-jpeg-interface/inc/mm_jpeg_dbg.h b/QCamera2/stack/mm-jpeg-interface/inc/mm_jpeg_dbg.h
index 4c0a162..116b57c 100755
--- a/QCamera2/stack/mm-jpeg-interface/inc/mm_jpeg_dbg.h
+++ b/QCamera2/stack/mm-jpeg-interface/inc/mm_jpeg_dbg.h
@@ -54,10 +54,10 @@
     #include <utils/Log.h>
     #define CDBG(fmt, args...) ALOGE(fmt, ##args)
   #else
-    #include <stdio.h>
-    #define CDBG(fmt, args...) fprintf(stderr, fmt"\n", ##args)
-    #define ALOGE(fmt, args...) fprintf(stderr, fmt"\n", ##args)
-    #define ALOGD(fmt, args...) fprintf(stderr, fmt"\n", ##args)
+    #include <syslog.h>
+    #define CDBG(fmt, args...) syslog(LOG_INFO, fmt, ##args)
+    #define ALOGE(fmt, args...) syslog(LOG_ERR, fmt, ##args)
+    #define ALOGD(fmt, args...) syslog(LOG_DEBUG, fmt, ##args)
   #endif
 #endif
 
@@ -65,7 +65,8 @@
   #define CDBG_HIGH(fmt, args...)  ALOGE(fmt, ##args)
   #define CDBG_ERROR(fmt, args...)  ALOGE(fmt, ##args)
 #else
-  #define CDBG_HIGH(fmt, args...) fprintf(stderr, fmt, ##args)
-  #define CDBG_ERROR(fmt, args...) fprintf(stderr, fmt, ##args)
+  #include <syslog.h>
+  #define CDBG_HIGH(fmt, args...) syslog(LOG_INFO, fmt, ##args)
+  #define CDBG_ERROR(fmt, args...) syslog(LOG_ERR, fmt, ##args)
 #endif
 #endif /* __MM_JPEG_DBG_H__ */
diff --git a/mm-image-codec/qomx_core/qomx_core.c b/mm-image-codec/qomx_core/qomx_core.c
index 7987e15..28ecfb6 100755
--- a/mm-image-codec/qomx_core/qomx_core.c
+++ b/mm-image-codec/qomx_core/qomx_core.c
@@ -34,7 +34,8 @@ IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.*/
 #ifdef _ANDROID_
 #include <utils/Log.h>
 #else
-#define ALOGE(fmt, args...) fprintf(stderr, fmt"\n", ##args)
+#include <syslog.h>
+#define ALOGE(fmt, args...) syslog(LOG_ERR, fmt, ##args)
 #define ALOGD(fmt, args...) do {} while(0)
 #endif
 
diff --git a/qcamlib/test/qcamlib_test.c b/qcamlib/test/qcamlib_test.c
index 19f5d89..99e86f8 100644
--- a/qcamlib/test/qcamlib_test.c
+++ b/qcamlib/test/qcamlib_test.c
@@ -34,12 +34,15 @@
 #include <stdbool.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/time.h>
 #include "qcamlib.h"
 
 #define DEF_DUMP_COUNT_MAX    100
 #define DEF_DUMP_INTERVAL     30
 #define DEF_OUTPUT_DIR        "camera"
 
+#define PRINT_FPS             0
+
 uint32_t frame_count = 0;
 uint32_t dump_count = 0;
 uint32_t dump_count_max = DEF_DUMP_COUNT_MAX;
@@ -50,6 +53,8 @@ bool previewing = false;
 
 char *outdir = DEF_OUTPUT_DIR;
 
+struct timeval tv0, tv1;
+
 char usage_str[] =
   "usage: qcamlib-test [options]\n"
   "\n"
@@ -68,6 +73,16 @@ void preview_cb(void* frame)
   int rc;
   frame_count++;
 
+#if PRINT_FPS
+  if (frame_count % 30 == 0) {
+    gettimeofday(&tv1, NULL);
+    double t_diff = (tv1.tv_sec - tv0.tv_sec) +
+      (tv1.tv_usec - tv0.tv_usec) / 1000000.0;
+    fprintf(stderr, "tv_diff = %f, fps = %f\n", t_diff, 30.0/t_diff);
+    tv0 = tv1;
+  }
+#endif
+
   if ((frame_count-1) % dump_interval == 0 && dump_count < dump_count_max) {
     rc = qcamlib_copy_preview_frame(buf, frame, bufsize);
     if (rc < 0) {
-- 
1.9.1

